hail/python/hailtop/aiocloud/common/credentials.py

```diff
@@ -2,15 +2,7 @@
 from typing import Dict
 
 
-class CloudCredentials(abc.ABC):
-    @staticmethod
-    def from_file(credentials_file):
-        raise NotImplementedError
-
-    @staticmethod
-    def default_credentials():
-        raise NotImplementedError
-
+class Credentials(abc.ABC):
```

Comment:
I think this is all annoyingly complicated because of `AnonymousCloudCredentials` where the `auth_headers` is empty. Maybe `AnonymousCloudCredentials` shouldn't exist and anything that can use anonymous access should accept an `Optional[Credentials]`.